### PR TITLE
build: fix docker compose arm and add ci

### PR DESCRIPTION
**Pull request by dohaki**
This allows to run the docker setup on ARM-based machines.

### CODE REVIEW

File: .github/workflows/lint-build-test.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Lint, build and test
+
+on:
+  push:
+    branches:
+      - stage
```
Comment (amateima) [inline .github/workflows/lint-build-test.yaml:6]: shouldn't master be added here as well? The base branch can change again after the PR has been made 🤔

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,6 @@
-FROM node:16-stretch AS development
+FROM node:16-alpine AS development
+
+RUN apk --no-cache add --virtual .builds-deps build-base python3 git
```
Comment (amateima) [inline Dockerfile:3]: I'm curious why stretch needs to be replaced with alpine. Is this related to the ARM issues you had?

File: docker-compose.yml [hunk 3]
```diff
@@ -33,23 +31,17 @@ services:
...
+    command: ["redis-server", "--appendonly", "yes"]
     environment:
       REDIS_PASSWORD: ${REDIS_PASSWORD}
-    networks:
-      - webnet
-    volumes: 
+    volumes:
       - scraper-api-redis-volume:/data
-networks:
-  webnet:
```
Comment (amateima) [inline docker-compose.yml:52]: Did you remove the network in order to allow the DB to be accessed by the BE app started outside `docker-compose`?

File: .github/workflows/lint-build-test.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name: Lint, build and test
```
Comment (amateima) [inline .github/workflows/lint-build-test.yaml:1]: I see the lint and build commands, but no testing command

File: .github/workflows/lint-build-test.yaml (hunk 1 above)
Comment (dohaki) [reply .github/workflows/lint-build-test.yaml:6]: Yea I can also add `master` here. But if you look at the line below, you can see `pull_request`. That means that on every PR for every base this workflow will run. In addition, we run on pushing / merging to `stage` via this `push` trigger.

File: Dockerfile (hunk 2 above)
Comment (dohaki) [reply Dockerfile:3]: Actually, it doesn't have to necessarily. The most important line is
```
RUN apk --no-cache add --virtual .builds-deps build-base python3 git
```

With stretch, I would need to change this to the `apt` equivalent I guess. I prefer `alpine` though because the image is way smaller (~ 5 MB) in contrast to stretch (~ 40 MB). Or is there any specific reason to use the `stretch` variant?

File: docker-compose.yml (hunk 3 above)
Comment (dohaki) [reply docker-compose.yml:52]: I had `ECONNREFUSED` errors in the BE container when started via `docker-compose`. I could fix it by removing this and using the default approach as described here https://docs.docker.com/compose/networking/.

File: .github/workflows/lint-build-test.yaml (hunk 4 above)
Comment (dohaki) [reply .github/workflows/lint-build-test.yaml:1]: Yea I played around with running the e2e tests as part of the pipeline but got stuck. I wanted to finish up the other PR first before continuing on that.

**Review by amateima — APPROVED** [review]

File: .github/workflows/lint-build-test.yaml (hunk 1 above)
Comment (dohaki) [reply .github/workflows/lint-build-test.yaml:6]: Ah ok yea then I will add master here. makes sense
